src/m365/teams/commands/user/user-app-remove.ts

```diff
@@ -110,12 +122,38 @@ class TeamsUserAppRemoveCommand extends GraphCommand {
     }
     else {
       const result = await Cli.promptForConfirmation({ message: `Are you sure you want to remove the app with id ${args.options.id} for user ${args.options.userId ?? args.options.userName}?` });
-
       if (result) {
         await removeApp();
       }
     }
   }
+
+  private async getAppId(args: CommandArgs): Promise<string> {
+    if (args.options.id) {
+      return args.options.id;
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/users/${args.options.userId}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/displayName eq '${formatting.encodeQueryParameter(args.options.name as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const response = await request.get<{ value: { id: string; }[] }>(requestOptions);
+    const app: { id: string; } | undefined = response.value[0];
+
+    if (!app) {
+      throw `The specified Teams app does not exist`;
+    }
+
+    if (response.value.length > 1) {
+      throw `Multiple Teams apps with name ${args.options.name} found. Please choose one of these ids: ${response.value.map(x => x.id).join(', ')}`;
```

Comment:
Rather than throwing an error here, we should use `Cli.handleMultipleResultsFound()` which shows a disambiguation prompt or throws error depending on a user's settings.